src/Worker.cpp

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright (c) 2021 Chiu Yen-Chen, Swen Sun-Yen, Wen Yong-Wei, Yuan Wei-Chen.
+All rights reserved.
+Use of this source code is governed by a BSD-style license that can be
+found in the LICENSE file. See the AUTHORS file for names of contributors.
+ */
+#ifndef SHORTLINK_SRC_WORKER_H_
+#define SHORTLINK_SRC_WORKER_H_
+#include <string>
+#include "Worker.h"
+#include "Parser.h"
+
+namespace shortlink
+{
+	//constructor
+	Worker::Worker()
+	{
+		worker_status = WorkerStatus.kFree;
```

Comment:
[Optional] Usually a worker has an associated thread, but we don't see that here, is this intentional? Maybe add a ThreadMain()?